corehq/apps/commtrack/models.py

```diff
@@ -503,20 +504,33 @@ def _yield_txns(section_id, quantity):
                     yield _txn(action=const.StockActions.RECEIPTS, case_id=dst,
                                section_id=section_id, quantity=quantity)
 
+        def _log_blank_quantity_error(config, section_id):
+            logging.error((
+                "Blank transaction quantity submitted on domain %s for "
+                "a %s ledger" % (config.domain, section_id)
+            ))
+
         section_id = action_node.attrib.get('section-id', None)
         grouped_entries = section_id is not None
         if grouped_entries:
-            quantity = float(product_node.attrib.get('quantity'))
-            for txn in _yield_txns(section_id, quantity):
-                yield txn
-
+            quantity = product_node.attrib.get('quantity')
+            # make sure quantity is not an empty, unset node value
+            if str(quantity).strip() != '':
```

Comment:
Would it make sense to check for other non-float values as well? Something like this:
```python
def float_or_none(value):
    try:
        return float(value)
    except (ValueError, TypeError):
        log.error("Blank transaction quantity submitted on domain %s for "
                  "a %s ledger", config.domain, section_id)
        return None

...

quantity = float_or_none(product_node.attrib.get('quantity'))
if quantity is not None:
    for txn in ...
```